event-hubs.api.md

```diff
@@ -0,0 +1,275 @@
+    enqueuedTime?: Date | number;
+    static fromEnqueuedTime(enqueuedTime: Date | number): EventPosition;
+    static fromOffset(offset: number, isInclusive?: boolean): EventPosition;
+    static fromSequenceNumber(sequenceNumber: number, isInclusive?: boolean): EventPosition;
+    isInclusive: boolean;
+    static latest(): EventPosition;
+    offset?: number | "@latest";
+    sequenceNumber?: number;
+    }
+
+// @public
+export interface GetEventHubPropertiesOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface GetPartitionIdsOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface GetPartitionPropertiesOptions extends SpanOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface LastEnqueuedEventProperties {
+    enqueuedOn?: Date;
+    offset?: string;
+    retrievalOn?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export interface PartitionManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<void>;
+}
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMs?: number;
+    ownerId: string;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
+        [key: string]: any;
+    };
+}
+
+export { RetryOptions }
+
+// @public
+export interface SendBatchOptions {
+    abortSignal?: AbortSignalLike;
+    parentSpan?: Span | SpanContext;
+}
+
+// @public
+export interface Subscription {
+    close(): Promise<void>;
+    isRunning: boolean;
+}
+
+export { TokenCredential }
+
+export { TokenType }
```

Comment:
Does this need to be exported? is it just for internal consumption?

Also I don't see the EventHubsSharedKeyCredential being exported.

Reply:
`TokenType` probably doesn't need to be.
SharedKeyCredential is not being exposed for GA (haven't seen a need for it yet, don't want to rush it).